app/modules/key_management/secret_manager.py

```
@@ -43,202 +65,557 @@ def _get_fernet() -> Fernet:
+                            )
+                            logger.info(
+                                f"Successfully created new secret in GCP for {service}"
+                            )
+                        except Exception as create_error:
+                            logger.error(
+                                f"Failed to create secret in GCP for {service}: {str(create_error)}"
+                            )
+                            raise HTTPException(
+                                status_code=500,
+                                detail=f"Failed to create secret: {str(create_error)}",
+                            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)